### ISSUE TITLE
POST api/order/ cancel returns error 400 bad_request when successful

### ISSUE THREAD

**Issue by jerryfletcher21** [RoboSats/robosats#1245]
**Describe the bug**
Other POST api/order/ `action` do not return error when successful, but `cancel` returns error 400 "bad_request": "This order has been cancelled by the maker". I think this is because when requesting an order that is canceled this is what is returned, but while doing the cancelling action is weird to be returned an error when successful. 
Maybe to fix, when requesting an order that is cancelled, it can be returned a json without error, something like "status": "This order has been cancelled by the maker".

in api/views.py:
```
        # 2) If order has been cancelled
        if order.status == Order.Status.UCA:
            return Response(
                {"bad_request": "This order has been cancelled by the maker"},
                status.HTTP_400_BAD_REQUEST,
            )
        if order.status == Order.Status.CCA:
            return Response(
                {"bad_request": "This order has been cancelled collaborativelly"},
                status.HTTP_400_BAD_REQUEST,
            )
```

**To Reproduce**
Make a POST request at api/order/ with the `action` `cancel`.

**Expected behavior**
Not to be returned an error when then cancel is successful.

**Comment by Reckless-Satoshi** [comment]
Indeed, `cancel_order` action will give you the same response as a GET /order, which will be a bad request because of cancelled order. This should be improved.

**Comment by KoalaSat** [comment]
Seems like a 204 does not accept a body on the response (which makes sense)

**Comment by keshav0479** [comment]
I’ve been digging into this and i think i found the cause.
It looks like the `cancel` action actually succeeds, but then the code falls through to call `self.get(request)`. Since the order is now marked "cancelled," that getter throws the 400 error we're seeing.

I’d love to take a shot at fixing this! My plan is to stop that fall-through and just return a `200 OK` JSON response immediately after the cancellation succeeds.

```python
# Proposed 
order.refresh_from_db()
return Response(
    {
        "status": "Order cancelled successfully", 
        "order_id": order.id, 
        "order_status": order.status
    }, 
    status.HTTP_200_OK
)

```

I noticed previous approach tried `204` but ran into issue with empty bodies. Does a simple `200 OK` sound like a better path forward? happy to open a PR if this looks good to you, also open to feedback if there's a better way to handle this

**Comment by KoalaSat** [comment]
@keshav0479 yes I think we can go with 200, please keep front-end in mind for your PR, thanks!
